src/test/java/bowlingexercise3/BowlingTest.java

```diff
@@ -61,22 +81,339 @@ public void testBowlingSpares(String turns, int score) {
+            "23|32|11|11|11|11|11|11|11|11||",
+            "45|54|11|11|11|11|11|11|11|11||",
+            "63|36|11|11|11|11|11|11|11|11||",
+            "72|27|11|11|11|11|11|11|11|11||",
+            "81|18|11|11|11|11|11|11|11|11||",
+            "81|18|11|11|11|11|11|11|11|11||",
+            "81|1-|-1|11|11|11|11|-1|11|11||",
+            "--|--|--|--|--|--|--|--|--|--||",
+            "22|22|22|22|22|22|22|22|22|22||",
+            "33|33|33|33|33|33|33|33|33|33||",
+            "44|44|44|44|44|44|44|44|44|44||",
+            "12|12|12|12|12|12|12|12|12|12||",
+            "21|21|21|21|21|21|21|21|21|21||",
+            "18|18|18|18|18|18|18|18|18|18||",
+            "72|72|72|72|72|72|72|72|72|72||",
+            "63|63|63|63|63|63|63|63|63|63||",
+            "54|54|54|54|54|54|54|54|54|54||",
+            "22|22|45|63|72|81|11|11|11|11||",
+            "11|54|11|11|11|11|11|11|11|11||",
+            "18|54|61|71|34|35|6-|-8|-4|17||",
+            "54|54|55|54|55|54|54|54|54|54||",
+            "45|45|45|45|45|45|45|45|45|45||",
+            "18|9-|9-|9-|9-|-9|-9|9-|9-|9-||",
+            "1/|1/|1/|1/|1/|1/|1/|1/|1/|1/||",
+            "-/|-/|-/|-/|-/|-/|-/|-/|-/|-/||",
+            "2/|2/|2/|2/|2/|2/|2/|2/|2/|2/||",
+            "3/|3/|3/|3/|3/|3/|3/|3/|3/|3/||",
+            "4/|4/|4/|4/|4/|4/|4/|4/|4/|4/||",
+            "5/|5/|5/|5/|5/|5/|5/|5/|5/|5/||",
+            "6/|6/|6/|6/|6/|6/|6/|6/|6/|6/||",
+            "7/|7/|7/|7/|7/|7/|7/|7/|7/|7/||",
+            "8/|8/|8/|8/|8/|8/|8/|8/|8/|8/||",
+            "9/|9/|9/|9/|9/|9/|9/|9/|9/|9/||",
+
+
+
+    })
+    public void testingValidInputNormalFrames(String turns) {
+
+        Bowling game = new Bowling();
+        Assertions.assertDoesNotThrow(() -> game.turns(turns));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+
+            "-/|-/|-/|-/|-/|-/|-/|-/|-/|-/||3",
+            "1/|-/|-/|-/|-/|-/|-/|-/|-/|-/||1",
+            "1/|1/|1/|1/|1/|1/|1/|1/|1/|1/||2",
+            "2/|2/|2/|2/|2/|2/|2/|2/|2/|2/||4",
+            "3/|3/|3/|3/|3/|3/|3/|3/|3/|3/||5",
+            "4/|4/|4/|4/|4/|4/|4/|4/|4/|4/||7",
+            "5/|5/|5/|5/|5/|5/|5/|5/|5/|5/||8",
+            "6/|6/|6/|6/|6/|6/|6/|6/|6/|6/||9",
+            "7/|7/|7/|7/|7/|7/|7/|7/|7/|7/||6",
+            "8/|8/|8/|8/|8/|8/|8/|8/|8/|8/||8",
+            "9/|9/|9/|9/|9/|9/|9/|9/|9/|9/||-",
+            "1/|1/|3/|4/|5/|6/|7/|7/|8/|9/||-",
+
+
+    })
+    public void testValidInputSpareBonus(String turns) {
+
+        Bowling game = new Bowling();
+        Assertions.assertDoesNotThrow(() -> game.turns(turns));
+    }
+
+    @ParameterizedTest
+    @CsvSource(value = {
+
+            "X|X|X|X|X|X|X|X|X|X||",
+            "X|11|11|11|11|11|11|11|11|11||",
+            "X|11|X|11|X|11|X|11|X|11||",
+            "X|11|11|11|X|X|X|X|X|X||",
+            "X|12|23|X|X|X|9-|X|X|X||",
+            "X|11|11|11|11|11|11|11|X|X||",
+
+
+
+    })
+    public void testValidInputStrikeNormal10Frames(String turns) {
```

Comment:
Is it testing strikes in all frames 1-10 - are these input strings valid? Some of them end in strikes, but don't have any bonus balls?

Reply:
I wanted to check whether strikes in any position work and are recognised as valid before I moved onto the bonus ball criteria.

Reply:
ok, but I thought it would throw an IllegalArgumentException if there were no bonus balls following a strike in the 10th frame?

Reply:
Updated